flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/result/ResultFetcher.java

```diff
@@ -213,11 +187,7 @@ public synchronized ResultSet fetchResults(long token, int maxFetchSize) {
                 throw new SqlExecutionException(msg);
             }
             return new ResultSet(
-                    ResultSet.ResultType.PAYLOAD,
-                    currentToken,
-                    resultSchema,
-                    converter,
-                    new LinkedList<>(bufferedPrevResults));
+                    ResultSet.ResultType.PAYLOAD, currentToken, resultSchema, bufferedPrevResults);
```

Comment:
Is the coping of `bufferedPrevResults` necessary? It seems there might be a new `fetchResult()` be called before the `ResultSet` is consumed.